src/main/java/htsjdk/variant/variantcontext/VariantContext.java

```diff
@@ -1233,7 +1233,21 @@ public void validateAlternateAlleles() {
             throw new TribbleException.InternalCodecException(String.format("one or more of the ALT allele(s) for the record at position %s:%d are not observed at all in the sample genotypes", getContig(), getStart()));
     }
 
+    private void validateNumberOfItems(final String attributeKey, final int observedSize ) {
+        if ( hasAttribute(attributeKey) && observedSize > 0) {
+            Object object = getAttribute(attributeKey);
+            int reportedSize = (object instanceof List ) ? ((List) object).size() : 1;
+            if (reportedSize != observedSize) {
+                throw new TribbleException.InternalCodecException(String.format("the %s tag has the incorrect number of records at position %s:%d, %d vs. %d", attributeKey, getContig(), getStart(), reportedSize, observedSize));
+            }
+        }
+    }
+
     public void validateChromosomeCounts() {
+        final int numberOfAlternateAlleles = alleles.size() - 1;
```

Comment:
you could reuse numberOfAlternateAlleles down below where `!getAlternateAlleles().isEmpty()` is checked to save an unnecessary List allocation